File: frontend/src/types/sach.ts
export type Sach = {
  maSach: string
  tieuDe: string
  tacGia: string
  namXuatBan?: number | null
  maTheLoai?: string | null
  theLoai?: string | null
  ngonNgu?: string | null
  tomTat?: string | null
  anhBiaUrl?: string | null
}

export type CreateSachPayload = {
  tieuDe: string
  tacGia: string
  namXuatBan?: number | null
  maTheLoai?: string | null
  ngonNgu?: string | null
  tomTat?: string | null
}

export type UpdateSachPayload = {
  tieuDe: string
  tacGia: string
  namXuatBan?: number | null
  maTheLoai?: string | null
  ngonNgu?: string | null
  tomTat?: string | null
}

File: MyWebAPI/Models/Sach.cs
namespace MyWebAPI.Models
{
    public class SachVM
    {
        public string? MaSach { get; set; }
        public string TenSach { get; set; } = default!;
        public string TacGia { get; set; } = default!;
        public int? NamXuatBan { get; set; }
        public string? TheLoai { get; set; }

        public string? NgonNgu { get; set; }

        public string? TomTat { get; set; }
    }
}
